### ISSUE TITLE
`vagrant ssh` does not work on Git Bash/MSYS2 environments after 2.4.0 update

### ISSUE THREAD

**Issue by guillermo-jano** [hashicorp/vagrant#13284]
After https://github.com/hashicorp/vagrant/pull/13219, looks like Vagrant now defaults to generating ed25519 key pairs when replacing the insecure key upon new box creation. On Windows, this is generated with CRLF line endings and apparently the OpenSSL version shipping with Git Bash and MSYS2 (probably also Cygwin, not tested) does not like that (this seemingly wasn't a problem with RSA keys though). `vagrant ssh` works as expected on Windows PowerShell and traditional Command Prompt (CMD.EXE).

### Debug output

```
$ vagrant ssh
vagrant@127.0.0.1's password:
```

If I try `vagrant ssh -- -v` instead I can see

```
Load key "C:/MYBOX/.vagrant/machines/default/virtualbox/private_key": error in libcrypto
```

Also if I run `dos2unix .vagrant/machines/default/virtualbox/private_key` and then run `vagrant ssh` again connection works as expected. So apparently the problem is the OpenSSL version in Git Bash/MSYS2 does not like ed25519 keys with CRLF line endings.

### Expected behavior

`vagrant ssh` should open a session with the newly created box without asking for a password

### Actual behavior

`vagrant ssh` asks for a password

#### Vagrant version

Vagrant 2.4.0

#### Host operating system

Microsoft Windows 10 Enterprise 10.0.19044 Build 19044

#### Guest operating system

Ubuntu 18.04.3 LTS

#### Steps to reproduce

1. Create a new Vagrant box with `vagrant up`
2. Run `vagrant ssh` from a Git Bash / MSYS2 environment (probably also Cygwin, not tested)

#### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "hashicorp/bionic64"
end
```

**Comment by iliareshetov** [comment]
I experienced a similar problem, which was resolved by switching CRLF to LF.

**Comment by guillermo-jano** [comment]
Yes the `dos2unix` I mentioned in the issue body fixes the problem, but IMO it would be better if this worked out of the box as it used to.

**Comment by iliareshetov** [comment]
Hey, I’m with you on this and I just want to confirm that this issue is there and it’s rather hard to figure out why it doesn’t work. 

The command ` vagrant ssh --debug` doesn’t help much and only when I used the `ssh `command and only then got `error in libcrypto`, which brought me here.

Definitely should work out of the box.

**Comment by guillermo-jano** [comment]
Yes, hopefully this gets some attention from the developers and can be fixed in a future release.

**Comment by mkot02** [comment]
I came across the same issue. I have setup with multiple hosts and ansible running in one of them. Ansible fails to do configuration on other hosts with message: `invalid format`.

As a workaround I'm converting private_keys using this command:
```
find .vagrant -type f -name private_key | xargs dos2unix
```

I have this problem after upgrading to 2.4.0

**Comment by EyalDagiAvdor** [comment]
Can confirm this happens in git bash on vagrant 2.4.0.
`vagrant ssh`
`Error: vagrant@127.0.0.1: Permission denied`

The workaround fixes the issue, but this should work out of the box.
`find .vagrant -type f -name private_key | xargs dos2unix`

**Comment by guillermo-jano** [comment]
I think this has been the case forever, however the SSH client in Git Bash / MSYS2 does not seem to have an issue with RSA private keys but it does not like ed25519 keys in CRLF format. Anyway since the built-in Windows SSH client is fine with LF keys, I'd say your solution should harm no one, and `vagrant ssh` would work in every kind of environment.

**Comment by ghalse** [comment]
> I think this has been the case forever, however the SSH client in Git Bash / MSYS2 does not seem to have an issue with RSA private keys but it does not like ed25519 keys in CRLF format.

It is a bit more subtle than that. The old RSA private keys were PEM encoded using OpenSSL; the ED25519 keys are encoded using OpenSSH's more lightweight format. Having looked at OpenSSH's source code for this, I think you'd have the same problem with an RSA key saved using the OpenSSH file format. (i.e. the changes in the way vagrant generates the keys might make this a problem for new RSA keys too).

**Comment by guillermo-jano** [comment]
Ahh, that might be, not sure how to ask Vagrant to generate a RSA key instead of ed25519 now anyway...

**Comment by chrisroberts** [comment]
I had not forgotten about this issue! The fix will be included in 2.4.1 :slightly_smiling_face:
